frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/DiscoverAction.java

```diff
@@ -56,11 +60,11 @@ public void actionPerformed(AnActionEvent anActionEvent) {
 
     Consumer<IGitMacheteRepositorySnapshot> saveAction = repositorySnapshot -> saveDiscoveredLayout(repositorySnapshot,
         GitVfsUtils.getMacheteFilePath(gitRepository), project, getGraphTable(anActionEvent),
-        getBranchLayoutWriter(anActionEvent));
-    Consumer<IGitMacheteRepositorySnapshot> saveAndEditAction = repositorySnapshot -> {
-      saveAction.accept(repositorySnapshot);
-      openMacheteFile(project, gitRepository);
-    };
+        getBranchLayoutWriter(anActionEvent), () -> {});
+    Consumer<IGitMacheteRepositorySnapshot> saveAndEditAction = repositorySnapshot -> saveDiscoveredLayout(repositorySnapshot,
```

Comment:
While you're here... I'd rather extract both those `Consumer`s to separate methods, this is hard to read in th ecurrent shape

Reply:
And then just pass them as `this::...` to `GraphTableDialog.Companion.of`